feat(prom/exp/snmp): add support to pass multiple SNMP config files to `prometheus.exporter.snmp` (#967)
grafana/alloy: docs/sources/reference/components/prometheus.exporter.snmp.md

The following arguments can be used to configure the exporter's behavior.
Omitted fields take their default values.

| Name          | Type                 | Description                                               | Default | Required |
| ------------- | -------------------- | --------------------------------------------------------- | ------- | -------- |
| `config_file` | `string`             | SNMP configuration file defining custom modules.          |         | no       |
| `config_files` | `list(string)`       | List of SNMP configuration files defining custom modules. |         | no       |
| `config`      | `string` or `secret` | SNMP configuration as inline string.                      |         | no       |

The `config_file` argument points to a YAML file defining which snmp_exporter modules to use.
Refer to [snmp_exporter](https://github.com/prometheus/snmp_exporter#generating-configuration) for details on how to generate a configuration file.

> Each file listed in in the `config_files` argument must conform to the `config_file` requirements.

Reviewer: ```suggestion
Each file listed in in the `config_files` argument must conform to the `config_file` requirements.

Only one of `config_file` or `config_files` can be provided.
```

> The `config_files` argument consist of list of YAML files with content as described as above for `config_file`.

Reviewer 2: ```suggestion
Each file listed in in the `config_files` argument must conform to the `config_file` requirements.
```
Could we simply to a more direct statement about the requirements?

Author: I don't think it's necessary since requirements for `config_files` are similar to `config_file` as above. It'd be a duplicate.

Reviewer 2: So then we should drop this sentence?
